### PR TITLE
Added Kotlin Spring Boot Container Engine Tutorial

### PR DESCRIPTION
**Pull request by hhariri**
- [x] Tech writing / styleguide review - @ccarpentiere 
- [x] Formatting review for publishing - @jmdobry

### CODE REVIEW

File: tutorials/kotlin-springboot-container-engine.md [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+    following content into it. Alternately, you can
+    [download](https://github.com/JetBrains/gcp-samples/blob/master/.dockerignore)
+    the Dockerfile file to study and customize.
+
+        .gradle
+        build
+        out
+    
+
+The Dockerfile performs a series of steps. It creates a `work` folder where sources will be copied to and built from.
```
Comment (ccarpentiere) [inline tutorials/kotlin-springboot-container-engine.md]: The Dockerfile performs a series of steps:
1. It creates a `work` folder where sources are copied to and built from.
2. It copies the sources from the local file system (which should be your project folder) to the container.
3. It sets the working directory to the `work` folder created and then runs the Gradle wrapper command (`gradlew`) to build the project. This wrapper automatically downloads Gradle and then builds the project. 
4. Once the project is built, the Dockerfile copies the output of the `jar` to the `work` folder, where it is ready for 
    execution when you invoke `docker run`.

File: tutorials/kotlin-springboot-container-engine.md [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+Visit [http://localhost:8080/message](http://localhost:8080/message) to see the application
+respond running locally from your Docker image.
+
+## Deploying your application
+
+Now you're ready to deploy your application to Kubernetes Engine!
+
+### Build the production image
+
+To deploy the app, you will use the
```
Comment (ccarpentiere) [inline tutorials/kotlin-springboot-container-engine.md]: To deploy the app, you will use [Container Builder](https://cloud.google.com/container-builder/) to build your Docker image in the cloud and store the resulting Docker image in your project in [Container Registry](https://cloud.google.com/container-registry/).

File: tutorials/kotlin-springboot-container-engine.md [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+[Google Cloud Container Registry](https://cloud.google.com/container-registry/).
+
+Execute the following command to run the build:
+
+    gcloud container builds submit --tag=gcr.io/${PROJECT_ID}/demo:v1 .
+
+Replace `${PROJECT_ID}` with the ID of your Google Cloud Platform project.
+The period at the end is required.
+
+After the build finishes, the image `gcr.io/${PROJECT_ID}/demo:v1` will be
```
Comment (ccarpentiere) [inline tutorials/kotlin-springboot-container-engine.md]: After the build finishes, the image `gcr.io/${PROJECT_ID}/demo:v1` is available.

File: tutorials/kotlin-springboot-container-engine.md [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+    This command creates a cluster of two machines. You can choose a different
+    size, but two is a good starting point.
+
+    It might take several minutes for the cluster to be created. You can check
+    the cloud console at http://cloud.google.com/console, under the Kubernetes
+    Engine section, to see that your cluster is running. You will also be able
+    to see the individual running VMs under the Compute Engine section. Note
+    that once the cluster is running, you will be charged for the VM usage.
+
+1.  Configure the gcloud command-line tool to use your cluster by default, so
```
Comment (ccarpentiere) [inline tutorials/kotlin-springboot-container-engine.md]: Configure the `gcloud` command-line tool to use your cluster by default, so you don't have to specify it every time for the remaining `gcloud` commands.

File: tutorials/kotlin-springboot-container-engine.md [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+
+### Deploy to the cluster
+
+A production deployment comprises two parts: your Docker container, and a
+front-end load balancer (which also provides a public IP address.)
+
+We'll assume that you built the image to `gcr.io/${PROJECT_ID}/demo:v1` and
+you've created the Kubernetes cluster as described above.
+
+1.  Create a deployment.
```
Comment (ccarpentiere) [inline tutorials/kotlin-springboot-container-engine.md]: Create a deployment:

File: tutorials/kotlin-springboot-container-engine.md [hunk 6]
```diff
@@ -0,0 +1,383 @@
...
+
+    This runs your image on a Kubernetes pod, which is the deployable unit in
+    Kubernetes. The pod opens port 8080, which is the port your Spring Boot
+    application is listening on.
+
+    You can view the running pods using:
+
+        kubectl get pods
+
+1.  Expose the application by creating a load balancer pointing at your pod.
```
Comment (ccarpentiere) [inline tutorials/kotlin-springboot-container-engine.md]: Expose the application by creating a load balancer pointing at your pod:

File: tutorials/kotlin-springboot-container-engine.md [hunk 7]
```diff
@@ -0,0 +1,383 @@
...
+        kubectl set image deployment/demo demo=gcr.io/${PROJECT_ID}/demo:v1
+
+**Note:** If a deployment gets stuck because an error in the image prevents
+it from starting successfuly, you can recover by undoing the rollout. See the
+[Kubernetes deployment documentation](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/)
+for more info.
+
+## Clean up
+
+After you've finished this tutorial, you can clean up the resources you created
```
Comment (ccarpentiere) [inline tutorials/kotlin-springboot-container-engine.md]: After you've finished this tutorial, clean up the resources you created on Google Cloud Platform so you won't be billed for them going forward. To clean up, either delete your Kubernetes Engine resources, or delete the entire project.

File: tutorials/kotlin-springboot-container-engine.md [hunk 8]
```diff
@@ -0,0 +1,383 @@
...
+        kubectl delete service demo
+
+1.  The load balancer will be deleted asynchronously. Wait for that process to
+    complete by monitoring the output of:
+
+        gcloud compute forwarding-rules list
+
+    The forwarding rule will disappear when the load balancer is deleted.
+
+1.  Delete the cluster, which deletes the resources used by the cluster,
```
Comment (ccarpentiere) [inline tutorials/kotlin-springboot-container-engine.md]: Delete the cluster, which deletes the resources used by the cluster, including virtual machines, disks, and network resources:

File: tutorials/kotlin-springboot-container-engine.md [hunk 9]
```diff
@@ -0,0 +1,383 @@
...
+    including virtual machines, disks, and network resources.
+
+        gcloud container clusters delete demo-cluster
+
+### Deleting the project
+
+Alternately, you can delete the project in its entirety. To do so using the
+gcloud tool, run:
```
Comment (ccarpentiere) [inline tutorials/kotlin-springboot-container-engine.md]: `gcloud`

File: tutorials/kotlin-springboot-container-engine.md [hunk 10]
```diff
@@ -0,0 +1,383 @@
...
+@RestController
+class MessageController {
+    @RequestMapping("/message")
+    fun message(): Message {
+        return Message("Hello from Google Cloud", "High")
+    }
+}
+```
+
+The package should match that of your group and artifact name.
```
[the remark this answers is not in the document]
Comment (hhariri) [reply tutorials/kotlin-springboot-container-engine.md]: It's assumed they know how the JVM/Java/Gradle/Spring work so it shouldn't be a problem.

**Comment by hhariri** [comment]
I've (hopefully) corrected all things pointed out. Just as a FYI, this tutorial was copied verbatim from the Elixir one, which I was told to base it on. The majority of the changes requested here would probably need to be applied to the original tutorial also, cause otherwise there will be discrepancies in style.

**Comment by hhariri** [comment]
Do I need to do something from my side for this?

**Pull request merged**